bsg_test/bsg_nonsynth_dpi_rom.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef __BSG_NONSYNTH_DPI_ROM_HPP
+#define __BSG_NONSYNTH_DPI_ROM_HPP
+#include <svdpi.h>
+#include <verilated.h>
+#include <bsg_nonsynth_dpi.hpp>
+
+extern "C" {
+        // DPI Export function: Get the value at an index in the
+        // rom. Calls $fatal if an invalid index is accessed.
+        extern svBitVecVal bsg_dpi_rom_get(int);
+}
+
+namespace bsg_nonsynth_dpi{
+
+        // dpi_rom is the C++ wrapper class for the
+        // bsg_nonsynth_dpi_rom verilog module.
+        // 
+        // Template Parameters:
+        // 
+        //   typename T: The C-type that this rom contains. While
```

Comment:
You might consider using https://en.cppreference.com/w/cpp/language/static_assert with https://en.cppreference.com/w/cpp/types#Type_traits_.28since_C.2B.2B11.29 to throw a compile time error if T is an invalid type.

Reply:
Can T be a struct?

Reply:
Hmmmnh. It could be. But I haven't tested that. There's a different DPI type for structs.

Reply:
Do you want me to use static assert?

Reply:
It was a suggestion.

Reply:
check this out: https://en.cppreference.com/w/cpp/types/is_class

Reply:
This would force a static assert (which is nice) but I'd still need to call the dpi_width constructor to force a runtime assert when there's a mismatch between verilog and C

So I'm leaning toward meh, though admitting its usefulness.

Also, #friday.